### ISSUE TITLE
Grid does not properly update scroll offsets when transitioning from no columns+rows to having some columns+rows if a scrollToRow or scrollToColumn is set.

### ISSUE THREAD

**Issue by bvaughn** [bvaughn/react-virtualized#218]
Steps to reproduce (using the [test harness](https://bvaughn.github.io/react-virtualized/?component=Grid)):

1. Set scroll-to-row to 100
1. Set number of rows and columns to 0
1. Set number of rows to 150
1. Set number of columns to 150

At this point the correct cells will be rendered but the `Grid`'s scroll-top will be 0 so the cells will not be visible. Scrolling slightly will correct things.

Note that reversing steps 3 and 4 above does not reproduce the problem.

**Comment by bvaughn** [comment]
The following test case can reproduce the behavior as well:
```js
it('should set the correct :scrollTop after row and column counts increase from 0', () => {
  render(getMarkup({
    columnCount: 0,
    rowCount: 150,
    scrollToRow: 100
  }))
  const grid = findDOMNode(render(getMarkup({
    columnCount: 150,
    rowCount: 150,
    scrollToRow: 100
  })))
  const expectedScrollTop = 100 * DEFAULT_ROW_HEIGHT - DEFAULT_HEIGHT + DEFAULT_ROW_HEIGHT
  expect(grid.scrollTop).toEqual(expectedScrollTop)
})
```

**Comment by bvaughn** [comment]
Resolved by c41b8cc
